### PR TITLE
Bug/mds 2418

### PR DESCRIPTION
**Pull request by mmumby**
- Updated permittee appointment dates to line up with new/historic amendments and amalgamations
- Updated the string when saving exploration permits on the FE
- updated some linting errors

### CODE REVIEW

File: services/core-api/app/api/mines/permits/permit/models/permit.py [hunk 1]
```diff
@@ -88,19 +88,17 @@ def validate_permit_no(self, key, permit_no):
         return permit_no
 
 
-from marshmallow_sqlalchemy import ModelConversionError, ModelSchema, ModelConverter, fields
+# from marshmallow_sqlalchemy import ModelConversionError, ModelSchema, ModelConverter, fields
```
Comment (nayr974) [inline services/core-api/app/api/mines/permits/permit/models/permit.py]: Is this change intentional? @Jsyro

File: services/core-api/app/api/mines/permits/permit_amendment/resources/permit_amendment.py [hunk 2]
```diff
@@ -62,24 +62,40 @@ def post(self, mine_guid, permit_guid, permit_amendment_guid=None):
...
         permittees = MinePartyAppointment.find_by_permit_guid(permit_guid)
+        if not permittees:
+            raise NotFound('Party appointments not found')
+
+        permit_issue_date = data.get('issue_date')
+        is_historical_permit = False
+
+        new_end_dates = MinePartyAppointment.find_appointment_end_dates(
+            permit_guid, permit_issue_date)
+        current_app.logger.debug(new_end_dates)
```
Comment (nayr974) [inline services/core-api/app/api/mines/permits/permit_amendment/resources/permit_amendment.py]: We should probably remove test code

File: services/core-api/app/api/mines/permits/permit_amendment/models/permit_amendment.py [hunk 3]
```diff
@@ -130,17 +130,15 @@ def validate_description(self, key, description):
...
-        ordered = True
-        include_fk = True
-        sqla_session = db.session
-        #model_converter = CoreConverter
-        exclude = ('create_user', 'create_timestamp', 'update_user', 'update_timestamp')
-
-
-PermitAmendment._schema = PermitAmendmentSchema
\ No newline at end of file
+# from marshmallow_sqlalchemy import ModelConversionError, ModelSchema, ModelConverter
```
Comment (nayr974) [inline services/core-api/app/api/mines/permits/permit_amendment/models/permit_amendment.py]: Is this change intentional? @Jsyro

File: services/core-api/app/api/mines/permits/permit_amendment/resources/permit_amendment.py (hunk 2 above)
Comment (mmumby) [reply services/core-api/app/api/mines/permits/permit_amendment/resources/permit_amendment.py]: oh good catch, missed that line

File: services/core-api/app/api/mines/permits/permit/models/permit.py (hunk 1 above)
Comment (mmumby) [reply services/core-api/app/api/mines/permits/permit/models/permit.py]: Yeah, commenting it out was intentional on my part. 

From my understanding, this was half working code, everything should be updated/working in his PR.  (just looked now and saw that his PR was merged, so I'll make sure to chat with him tomorrow)

Comment (mmumby) [reply services/core-api/app/api/mines/permits/permit/models/permit.py]: removed

File: services/core-api/app/api/mines/permits/permit_amendment/models/permit_amendment.py (hunk 3 above)
Comment (mmumby) [reply services/core-api/app/api/mines/permits/permit_amendment/models/permit_amendment.py]: removed as per discussion with Jason.

File: services/core-web/src/components/mine/Compliance/ComplianceOrdersTable.js [hunk 4]
```diff
@@ -71,6 +71,15 @@ const fileColumns = [
   },
 ];
 
+const transformFileRowData = (file, inspectionId) => ({
```
Comment (pdittaro) [inline services/core-web/src/components/mine/Compliance/ComplianceOrdersTable.js]: This `const` is already declared earlier in this file, its causing issues on the build pipeline.

**Comment by pdittaro** [comment]
There is a failing unit test on Mine Party Appointments. It looks like this is related to the way that the Permit factory is being constructed.
